const userService = require('../services/userService');
const { EMAIL_ALREADY_TAKEN } = require('../core/errors');

async function createUser(req, res, next) {
    const { email, name, password } = req.body;
    try {
      await userService.checkEmailExistence(email);

        res.status(201).json({ message: 'User created successfully' });
    } catch (error) {
        if (error === EMAIL_ALREADY_TAKEN) {
            res.status(409).json({ error: 'Email already taken' });
        } else {
            res.status(500).json({ error: 'Internal server error' });
        }
    }
}

module.exports = {
    createUser,
};

